### PR TITLE
Add support for SSH connection debugging.

### PR DESCRIPTION
**Pull request by rhass**
This adds support for SSH connection debugging. Valid options are, `debug`, `info`, `warn`, `error`, and `fatal`.

Details about this option and others can be found here:
https://net-ssh.github.io/ssh/v2/api/classes/Net/SSH.html

### CODE REVIEW

File: lib/kitchen/transport/ssh.rb [hunk 1]
```diff
@@ -326,7 +326,8 @@ def connection_options(data) # rubocop:disable Metrics/MethodLength
           :timeout                => data[:connection_timeout],
           :connection_retries     => data[:connection_retries],
           :connection_retry_sleep => data[:connection_retry_sleep],
-          :max_wait_until_ready   => data[:max_wait_until_ready]
+          :max_wait_until_ready   => data[:max_wait_until_ready],
+          :verbose                => data[:verbose].to_sym
```
Comment (mwrock) [inline lib/kitchen/transport/ssh.rb]: I think the failed tests are telling you that you need a `nil` check

**Comment by rhass** [comment]
I managed to make RupoCop angry with that fix.

**Comment by jorhett** [comment]
would love to see this merged...

**Comment by cheeseplus** [comment]
+1

**Comment by mwrock** [comment]
:+1:

**Pull request merged**